### PR TITLE
ath79: add support for Ditel DT-ACU100

### PR DESCRIPTION
**Pull request by lsgxeva**
Ditel DT-ACU100 is a ceiling mount Router with PoE support, based on Qualcomm/Atheros QCA9531@650MHz SoC.

Short specification:

- QCA9531 650 MHz SoC
- 1x 10/100/1000 Mbps Ethernet (wan/eth1)
- 1x 10/100/1000 Mbps Ethernet (lan/eth0)
- 128MB of DDR2 RAM (Winbond W971GG6SB)
- 16 MB of FLASH (Winbond W25Q128FV)
- 2T2R 2.4 GHz, 802.11b/g/n (wlan2g)
- 1x RGB LED, 1x button
- UART header on PCB with proper labelling

Initial flashing instructions:

Original firmware is based on OpenWrt.

a) Use sysupgrade image directly in vendor GUI.

b) Or via tftp:

  ipaddr=192.168.1.1 serverip=192.168.1.10 bootfile="firmware.bin"

c) Or possibly via u-boot's `httpd` command.

MAC-address mapping follows original firmware:

* eth1 (wan) is the lowest mac address (art @ 0x0)
* eth0 (lan) uses eth1 + 1 (art @ 0x1000)
* wlan2g (phy0) uses eth1 + 2 (art @ 0x06)
* unused MAC (eth1 + 3) (art @ 0x5006)

Art dump (`hexdump /dev/mtd1 |grep ZZZZ`):

0000000 ZZZZ XXXX XXX0 ZZZZ XXXX XXX2 ffff ffff
0001000 0202 ZZZZ XXXX XXX1 0000 0000 0000 0000
0005000 202f bd21 0101 ZZZZ XXXX XXX3 0000 2000

flash Stock :

0x000000-0x040000(u-boot)
00x000000-0x050000(u-boot-env)
0x050000-0xfe0000(firmware)
0xfe0000-0xff0000(ditel)
0xff0000-0x1000000(art)

dev:    size   erasesize  name
mtd0: 00040000 00010000 "u-boot"
mtd1: 00010000 00010000 "u-boot-env"
mtd2: 00f90000 00010000 "firmware"
mtd3: 00290000 00010000 "kernel"
mtd4: 00d00000 00010000 "rootfs"
mtd5: 00190000 00010000 "rootfs_data"
mtd6: 00010000 00010000 "ditel"
mtd7: 00010000 00010000 "art"

Root access to original firmware (only via UART) can be achieved by making a backup of configuration from web interface. Backup contains whole `/etc` directory...

Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

### CODE REVIEW

File: package/boot/uboot-envtools/files/ath79 [hunk 1]
```diff
@@ -83,6 +83,7 @@ yuncore,a782|\
 yuncore,a930|\
 yuncore,xd3200|\
 yuncore,xd4200|\
+ditel,dt-acu100|\
```
Comment (musashino205) [inline package/boot/uboot-envtools/files/ath79:86]: Please sort alphabetically.

File: target/linux/ath79/dts/qca9531_ditel_dt-acu100.dts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	leds {
+		compatible = "gpio-leds";
+
+		led_status: status {
+			function = LED_FUNCTION_STATUS;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+
+		led_wlan: wlan {
```
Comment (musashino205) [inline target/linux/ath79/dts/qca9531_ditel_dt-acu100.dts:42]: Please remove unused DT label (`led_wlan: `).

File: target/linux/ath79/dts/qca9531_ditel_dt-acu100.dts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+
+	i2c: i2c {
+		compatible = "i2c-gpio";
+
+		sda-gpios = <&gpio 3 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+		scl-gpios = <&gpio 15 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+	};
```
Comment (musashino205) [inline target/linux/ath79/dts/qca9531_ditel_dt-acu100.dts:55]: What is this for?

File: target/linux/ath79/dts/qca9531_ditel_dt-acu100.dts [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+				};
+			};
+		};
+	};
+
+	spidev@1 {
+		compatible = "spidev";
+		reg = <1>;
+		spi-max-frequency = <12000000>;
+	};
```
Comment (musashino205) [inline target/linux/ath79/dts/qca9531_ditel_dt-acu100.dts:125]: ditto

File: target/linux/ath79/generic/base-files/etc/board.d/01_leds [hunk 5]
```diff
@@ -540,6 +540,9 @@ wd,mynet-wifi-rangeextender)
 xiaomi,aiot-ac2350)
 	ucidef_set_led_switch "wan" "WAN" "blue:wan" "switch0" "0x02"
 	;;
+ditel,dt-acu100)
+	ucidef_set_led_netdev "wan" "WAN" "red:wan" "eth0"
+	;;
```
Comment (musashino205) [inline target/linux/ath79/generic/base-files/etc/board.d/01_leds:545]: `eth1` is set as WAN in **02_network**, but `eth0` is used for "WAN" LED here...

File: target/linux/ath79/image/generic.mk [hunk 6]
```diff
@@ -3237,6 +3237,17 @@ define Device/xiaomi_mi-router-4q
...
 TARGET_DEVICES += xiaomi_mi-router-4q
 
+define Device/ditel_dt-acu100
+  SOC := qca9531
+  DEVICE_VENDOR := Ditel
+  DEVICE_MODEL := DT-ACU100
+  DEVICE_PACKAGES := -kmod-ath9k -uboot-envtools kmod-usb2
+  IMAGE_SIZE := 15936k
+  LOADER_FLASH_OFFS := 0x50000
+  SUPPORTED_DEVICES += dt-acu100
```
Comment (musashino205) [inline target/linux/ath79/image/generic.mk:3247]: Is this additional `SUPPORTED_DEVICES` necessary?

File: target/linux/ath79/image/generic.mk [hunk 7]
```diff
@@ -3237,6 +3237,17 @@ define Device/xiaomi_mi-router-4q
 endef
 TARGET_DEVICES += xiaomi_mi-router-4q
 
+define Device/ditel_dt-acu100
+  SOC := qca9531
+  DEVICE_VENDOR := Ditel
+  DEVICE_MODEL := DT-ACU100
+  DEVICE_PACKAGES := -kmod-ath9k -uboot-envtools kmod-usb2
```
Comment (musashino205) [inline target/linux/ath79/image/generic.mk:3244]: Isn't `kmod-ath9k` necessary for the wmac on QCA9531?

File: target/linux/ath79/image/generic.mk [hunk 8]
```diff
@@ -3237,6 +3237,17 @@ define Device/xiaomi_mi-router-4q
...
+  IMAGE_SIZE := 15936k
+  LOADER_FLASH_OFFS := 0x50000
```
Comment (musashino205) [inline target/linux/ath79/image/generic.mk:3246]: `LOADER_FLASH_OFFS` is not used for your device, please remove.

File: package/boot/uboot-envtools/files/ath79 (hunk 1 above)
Comment (lsgxeva) [reply package/boot/uboot-envtools/files/ath79:86]: okay, i will modify it.

File: target/linux/ath79/dts/qca9531_ditel_dt-acu100.dts (hunk 2 above)
Comment (lsgxeva) [reply target/linux/ath79/dts/qca9531_ditel_dt-acu100.dts:42]: Okay, i will remove led_wlan

File: target/linux/ath79/dts/qca9531_ditel_dt-acu100.dts (hunk 3 above)
Comment (lsgxeva) [reply target/linux/ath79/dts/qca9531_ditel_dt-acu100.dts:55]: Okay, i will remove i2c-gpio

File: target/linux/ath79/generic/base-files/etc/board.d/01_leds (hunk 5 above)
Comment (lsgxeva) [reply target/linux/ath79/generic/base-files/etc/board.d/01_leds:545]: Okay, i will change eth0 to eth1

File: target/linux/ath79/image/generic.mk (hunk 7 above)
Comment (lsgxeva) [reply target/linux/ath79/image/generic.mk:3244]: Okay, i will change to "DEVICE_PACKAGES := kmod-usb2"

File: target/linux/ath79/image/generic.mk (hunk 8 above)
Comment (lsgxeva) [reply target/linux/ath79/image/generic.mk:3246]: Okay, i will remove it.

File: target/linux/ath79/image/generic.mk (hunk 6 above)
Comment (lsgxeva) [reply target/linux/ath79/image/generic.mk:3247]: Okay, i will remove it.

File: target/linux/ath79/dts/qca9531_ditel_dt-acu100.dts (hunk 3 above)
Comment (lsgxeva) [reply target/linux/ath79/dts/qca9531_ditel_dt-acu100.dts:55]: to support tps23861

Comment (musashino205) [reply target/linux/ath79/dts/qca9531_ditel_dt-acu100.dts]: If it is actually used in hardware, explicitly describe a child node for that chip.

https://elixir.bootlin.com/linux/v6.11.5/source/Documentation/devicetree/bindings/hwmon/ti,tps23861.yaml

File: target/linux/ath79/generic/base-files/etc/board.d/01_leds [hunk 9]
```diff
@@ -540,6 +540,9 @@ wd,mynet-wifi-rangeextender)
 xiaomi,aiot-ac2350)
 	ucidef_set_led_switch "wan" "WAN" "blue:wan" "switch0" "0x02"
 	;;
+ditel,dt-acu100)
+	ucidef_set_led_netdev "wan" "WAN" "red:wan" "eth1"
```
Comment (raenye) [inline target/linux/ath79/generic/base-files/etc/board.d/01_leds:544]: What LED is `red:wan`? it's not referenced above.

**Comment by raenye** [comment]
Your PR should consist of a single commit. Please squash the commits and push-force (do not use merge, only rebase).

**Pull request closed without merging**